// nuxt-calendly/dist/runtime/composables/useCalendlyEventListener.mjs
import { onMounted, onUnmounted } from "vue";
export var CalendlyEvent = /* @__PURE__ */ ((CalendlyEvent2) => {
  CalendlyEvent2["PROFILE_PAGE_VIEWED"] = "calendly.profile_page_viewed";
  CalendlyEvent2["EVENT_TYPE_VIEWED"] = "calendly.event_type_viewed";
  CalendlyEvent2["DATE_AND_TIME_SELECTED"] = "calendly.date_and_time_selected";
  CalendlyEvent2["EVENT_SCHEDULED"] = "calendly.event_scheduled";
  CalendlyEvent2["PAGE_HEIGHT"] = "calendly.page_height";
  return CalendlyEvent2;
})(CalendlyEvent || {});
const EVENT_NAME = "message";
export default function useCalendlyEventListener(eventHandlers) {
  const { onDateAndTimeSelected, onEventScheduled, onEventTypeViewed, onProfilePageViewed, onPageHeightResize } = eventHandlers || {};
  const onMessage = (e) => {
    const eventName = e.data.event;
    if (eventName === "calendly.date_and_time_selected" /* DATE_AND_TIME_SELECTED */) {
      onDateAndTimeSelected && onDateAndTimeSelected(e);
    } else if (eventName === "calendly.event_scheduled" /* EVENT_SCHEDULED */) {
      onEventScheduled && onEventScheduled(e);
    } else if (eventName === "calendly.event_type_viewed" /* EVENT_TYPE_VIEWED */) {
      onEventTypeViewed && onEventTypeViewed(e);
    } else if (eventName === "calendly.profile_page_viewed" /* PROFILE_PAGE_VIEWED */) {
      onProfilePageViewed && onProfilePageViewed(e);
    } else if (eventName === "calendly.page_height" /* PAGE_HEIGHT */) {
      onPageHeightResize && onPageHeightResize(e);
    }
  };
  onMounted(() => {
    if (!import.meta.client)
      return;
    window.addEventListener(EVENT_NAME, onMessage);
  });
  onUnmounted(() => {
    if (!import.meta.client)
      return;
    window.removeEventListener(EVENT_NAME, onMessage);
  });
}
